core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -1058,27 +1121,52 @@ public void week() {
     testWeek("1999-12-31", 0, 52);
   }
 
+  @Test
+  public void testWeekOfYearWithTimeType() {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression expression = DSL
+        .week(functionProperties, DSL.literal(new ExprTimeValue("12:23:34")), DSL.literal(0));
+    assertEquals(INTEGER, eval(expression).type());
+    assertEquals(LocalDate.now(
+            functionProperties.getQueryStartClock()).get(ALIGNED_WEEK_OF_YEAR),
+        eval(expression).integerValue());
+    assertEquals("week(TIME '12:23:34', 0)", expression.toString());
+
+    expression = DSL.week_of_year(
+        functionProperties,
+        DSL.literal(new ExprTimeValue("12:23:34")));
+    assertEquals(INTEGER, eval(expression).type());
+    assertEquals(LocalDate.now(
+            functionProperties.getQueryStartClock()).get(ALIGNED_WEEK_OF_YEAR),
+        eval(expression).integerValue());
+    assertEquals("week_of_year(TIME '12:23:34')", expression.toString());
+  }
+
   @Test
   public void modeInUnsupportedFormat() {
     testNullMissingWeek(DATE);
 
     FunctionExpression expression1 = DSL
-        .week(DSL.literal(new ExprDateValue("2019-01-05")), DSL.literal(8));
+        .week(functionProperties, DSL.literal(new ExprDateValue("2019-01-05")), DSL.literal(8));
     SemanticCheckException exception =
         assertThrows(SemanticCheckException.class, () -> eval(expression1));
     assertEquals("mode:8 is invalid, please use mode value between 0-7",
         exception.getMessage());
 
     FunctionExpression expression2 = DSL
-        .week(DSL.literal(new ExprDateValue("2019-01-05")), DSL.literal(-1));
+        .week(functionProperties, DSL.literal(new ExprDateValue("2019-01-05")), DSL.literal(-1));
     exception = assertThrows(SemanticCheckException.class, () -> eval(expression2));
     assertEquals("mode:-1 is invalid, please use mode value between 0-7",
         exception.getMessage());
   }
 
   private void testWeekOfYear(String date, int mode, int expectedResult) {
```

Comment:
we really shouldn't call all the functions `test...`.  The helper functions should be called something else.

Reply:
Fixed in 8eea6302b9a2e22afd74879d5b1b06ef684b3d64